ultravox/tools/ds_tool.py

```diff
@@ -50,10 +53,16 @@ class TextGenerationTask:
     template: str = simple_parsing.field(default=DEFAULT_TEXTGEN_TEMPLATE, alias="-T")
 
     language_model: str = simple_parsing.field(default="gpt-4o", alias="-m")
+    base_url: str = simple_parsing.field(
+        default="https://api.openai.com/v1", alias="-b"
+    )
+    api_key: str = simple_parsing.field(default=None, alias="-k")
     max_tokens: int = 128
     temperature: float = 0
 
     def __post_init__(self):
+        global chat_client
+        chat_client = openai.Client(base_url=self.base_url, api_key=self.api_key)
```

Comment:
nit: assigning to global vars seems like a bad smell but I know why you had to do it.
I suggest putting a comment here that explains the reasoning: e.g. multi-processing not allowing non-serializable objects.